src/lib.rs

```diff
@@ -200,6 +201,208 @@ impl ReturnMode {
+        match id {
+            0 => SegmentationTypeId::NotIndicated,
+            1 => SegmentationTypeId::ContentIdentification,
+            16 => SegmentationTypeId::ProgramStart,
+            17 => SegmentationTypeId::ProgramEnd,
+            18 => SegmentationTypeId::ProgramEarlyTermination,
+            19 => SegmentationTypeId::ProgramBreakaway,
+            20 => SegmentationTypeId::ProgramResumption,
+            21 => SegmentationTypeId::ProgramRunoverPlanned,
+            22 => SegmentationTypeId::ProgramRunoverUnplanned,
+            23 => SegmentationTypeId::ProgramOverlapStart,
+            24 => SegmentationTypeId::ProgramBlackoutOverride,
+            25 => SegmentationTypeId::ProgramStartInProgress,
+            32 => SegmentationTypeId::ChapterStart,
+            33 => SegmentationTypeId::ChapterEnd,
+            34 => SegmentationTypeId::BreakStart,
+            35 => SegmentationTypeId::BreakEnd,
+            48 => SegmentationTypeId::ProviderAdvertisementStart,
+            49 => SegmentationTypeId::ProviderAdvertisementEnd,
+            50 => SegmentationTypeId::DistributorAdvertisementStart,
+            51 => SegmentationTypeId::DistributorAdvertisementEnd,
+            52 => SegmentationTypeId::ProviderPlacementOpportunityStart,
+            53 => SegmentationTypeId::ProviderPlacementOpportunityEnd,
+            54 => SegmentationTypeId::DistributorPlacementOpportunityStart,
+            55 => SegmentationTypeId::DistributorPlacementOpportunityEnd,
+            64 => SegmentationTypeId::UnscheduledEventStart,
+            65 => SegmentationTypeId::UnscheduledEventEnd,
+            80 => SegmentationTypeId::NetworkStart,
+            81 => SegmentationTypeId::NetworkEnd,
+            _ => SegmentationTypeId::Reserved
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum SegmentationUpid {
+    None,
+    SegmentationUpid {
+        upid: Vec<u8>
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Flag {
+    NotSet,
+    Set
+}
+impl Flag {
+    pub fn from_flag(flag: bool) -> Flag {
+        match flag {
+            false => Flag::NotSet,
+            true => Flag::Set
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum DeviceRestrictions {
+    RestrictGroup0,
+    RestrictGroup1,
+    RestrictGroup2,
+    None
+}
+impl DeviceRestrictions {
+    /// panics if `id` is something other than `0` or `1`
+    pub fn from_bits(restriction: u8) -> DeviceRestrictions {
+        match restriction {
+            0 => DeviceRestrictions::RestrictGroup0,
+            1 => DeviceRestrictions::RestrictGroup1,
+            2 => DeviceRestrictions::RestrictGroup2,
+            3 => DeviceRestrictions::None,
+            _ => panic!(
+                "Invalid device_restrictions value: {} (expected 0, 1, 2, 3)",
+            restriction),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum DeliveryRestrictionFlags {
+    None,
+    DeliveryRestrictions {
+        web_delivery_allowed_flag: Flag,
+        no_regional_blackout_flag: Flag,
```

Comment:
For these flags, I'd prefer either just a `bool`, or the more verbose option of a per-flag enum so that the caller can understand the meaning purely from the enum-varient,

e.g.

```
enum RegionalBlackout {
    Blackout,
    NoBlackout,
}
```

then this code would become,

```suggestion
        regional_blackout: RegionalBlackout,
```

Kinda a matter of taste though!

Reply:
I opted for `bool` and retained the original names since.